helm/stargate/templates/stargate_coordinator_deployment.yaml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: stargate-coordinator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: coordinator
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: coordinator
+  template:
+    metadata:
+      labels:
+        app: coordinator
+    spec:
+      containers:
+      - name: coordinator
+        {{ if eq .Values.cassandra.clusterVersion "3.11" }}
+        image: "{{ .Values.image.repository311 }}:{{ .Values.image.tag }}" 
+        {{ else if eq .Values.cassandra.clusterVersion "6.8" }}
+        image: "{{ .Values.image.repositoryDse68 }}:{{ .Values.image.tag }}"
+        {{ else }}
+        image: "{{ .Values.image.repository40 }}:{{ .Values.image.tag }}"
+        {{end}}
+        ports:
+        - containerPort: 8091
+        - containerPort: 8090
+        - containerPort: 9042
+        - containerPort: 8081
+        args:          
+         {{  range .Values.arguments }}
+           - {{ . }}
+         {{ end }}
+        readinessProbe:
+          httpGet:
+            path: /checker/readiness
+            port: 8084
+          timeoutSeconds: 10
+          periodSeconds: 10
+          failureThreshold: 30
+          initialDelaySeconds: 30
+        livenessProbe:
+          httpGet:
+            path: /checker/liveness
+            port: 8084
+          timeoutSeconds: 10
+          periodSeconds: 10
+          failureThreshold: 5
+          initialDelaySeconds: 300
+        resources:
+          limits:
+            cpu: {{ mul 4 (.Values.cpuReqMillicores | default 1000) }}m
```

Comment:
It seems to be fairly common across charts to define default resources as a yaml block in values.yaml and then in the template it would look something like this:

```
          resources:
            {{- toYaml .Values.resources | nindent 12 }}
```
This provides flexibility for users and it keeps the template logic simpler.